support/vm/vm.go

```diff
@@ -64,6 +66,50 @@ type InternalMessage struct {
 	params interface{}
 }
 
+type ChainMessage struct {
+	Version uint64
+
+	From address.Address
+	To   address.Address
+
+	Nonce uint64
+
+	Value abi.TokenAmount
+
+	GasLimit   int64
+	GasFeeCap  abi.TokenAmount
+	GasPremium abi.TokenAmount
+
+	Method abi.MethodNum
+	Params []byte
+}
+
+func makeChainMessage(from, to address.Address, nonce uint64, value abi.TokenAmount, method abi.MethodNum, params interface{}) (*ChainMessage, error) {
+	var buf bytes.Buffer
+	if params == nil {
+		if err := abi.Empty.MarshalCBOR(&buf); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := params.(cbor.Er).MarshalCBOR(&buf); err != nil {
+			return nil, err
+		}
+	}
+
+	return &ChainMessage{
+		Version:    0,
+		From:       from,
+		To:         to,
+		Nonce:      nonce,
+		Value:      value,
+		GasLimit:   5_000_000_000,
+		GasFeeCap:  big.Zero(),
+		GasPremium: big.Zero(),
```

Comment:
I was guessing that the GasLimit above was set non-zero so as to better represent the serialized size of a chain message. If so, these should be non-zero too. If not, just set the unused GasLimit to zero to match

Reply:
Yes GasLimit is set non-zero for serialization accuracy purposes. We can't set it to zero because then we will hit SysErrOutOfGas immediately.  If we set GasFeeCap and GasPremium non-zero we will have to implement base fee burning and miner payments which I want to avoid.